### PR TITLE
feat(city): create base entity and crud operations

### PR DESCRIPTION
**Pull request by cchampou**

### CODE REVIEW

File: src/modules/city/city.service.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { InjectRepository } from '@nestjs/typeorm';
+import { City } from './entities/city.entity';
+import { Repository } from 'typeorm';
+import { Country } from '../country/entities/country.entity';
+
+@Injectable()
+export class CityService {
+  constructor(
+    @InjectRepository(City) private cityRepository: Repository<City>,
+    @InjectRepository(Country) private countryRepository: Repository<Country>,
```
Comment (draphikas) [inline src/modules/city/city.service.ts:13]: Is this necessary ?

File: src/modules/country/entities/country.entity.ts [hunk 2]
```diff
@@ -9,4 +10,7 @@ export class Country {
     unique: true,
   })
   name: string;
+
+  @OneToMany(() => City, (city) => city.country)
+  cities: City[];
```
Comment (draphikas) [inline src/modules/country/entities/country.entity.ts:15]: Following typeorm doc, we need to add Relation type on both sides

File: src/modules/city/city.module.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
+import { CityService } from './city.service';
+import { CityController } from './city.controller';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { City } from './entities/city.entity';
+import { Country } from '../country/entities/country.entity';
+
+@Module({
+  imports: [TypeOrmModule.forFeature([City, Country])],
```
Comment (draphikas) [inline src/modules/city/city.module.ts:9]: Based on the other comment, if we only use one repository, we don't need to set Country in feature

File: src/modules/city/entities/city.entity.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+@Entity()
+export class City {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  name: string;
+
+  @ManyToOne(() => Country, (country) => country.cities)
+  country: Relation<Country>;
```
Comment (draphikas) [inline src/modules/city/entities/city.entity.ts:13]: Maybe set to NOT NULL ? I don't know a city without a country

File: src/modules/city/dto/update-city.dto.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { CreateCityDto } from './create-city.dto';
+
+export class UpdateCityDto extends CreateCityDto {}
```
Comment (draphikas) [inline src/modules/city/dto/update-city.dto.ts:3]: So, do we add id in DTO ?

File: src/modules/city/dto/create-city.dto.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import { Country } from '../../country/entities/country.entity';
+
+export class CreateCityDto {
+  @IsNotEmpty()
+  @IsString()
+  name: string;
+
+  @IsObject()
+  @IsOptional()
+  country: Partial<Country>;
```
Comment (draphikas) [inline src/modules/city/dto/create-city.dto.ts:11]:
```suggestion
  country: Pick<Country, "id">;
```

File: src/modules/city/city.service.ts (hunk 1 above)
Comment (cchampou) [reply src/modules/city/city.service.ts:13]: Not anymore, fixing !

File: src/modules/country/entities/country.entity.ts (hunk 2 above)
Comment (Rossb0b) [reply src/modules/country/entities/country.entity.ts:15]: like 
```ts
@Entity()
export class Webhook {
  @OneToMany(() => WebhookLog, (table) => table.webhook, { nullable: true })
  logs: WebhookLog[];
 }
 
@Entity()
export class WebhookLog {
  @ManyToOne(() => Webhook, (table) => table.logs, { nullable: true, onDelete: "CASCADE" })
  @JoinColumn({ name: "webhook_id" })
  webhook: Webhook;
  @Column("uuid", { nullable: true })
  webhookId: string;
}
```

**Review by Rossb0b — APPROVED** [review]

**Pull request merged**